### ISSUE TITLE
uAgent almanac tracking

### ISSUE THREAD

**Issue by bryanchriswhite** [fetchai/ledger-subquery#176]
## Acceptance criteria

1. Almanac registration events are tracked by a [respective entity](https://github.com/fetchai/contract-agent-almanac/issues/1).
2. Live almanac registrations must be distinct from historic ones and **must exclude expired** registrations.

**Comment by ejfitzgerald** [comment]
Makes sense to me the only thing to consider is that with the new version of the contract API

```graphql
type AlmanacService @jsonField {
  protocols: [String!]
  endpoints: [String!]!
}
```

becomes more like



```graphql
type AlmanacEndpoint @jsonField {
  endpoint: String!
  weight: Int!
}

type AlmanacService @jsonField {
  protocols: [String!]
  endpoints: [AlmanacEndpoint!]!
}
```

**Comment by bryanchriswhite** [comment]
I ran into some unexpected complications regarding foreign key constraints when attempting to relate `AlmanacRegistration` to `Event`. My current hypothesis is that `handleEvent` may not have finished at the time which `handleAlmanacRegistration` is attempting to save the registration, resulting in a foreign key constraint violation as the related event hasn't been committed yet.

I also experienced an issue while trying to add the reverse relationship between `AlmanacResolution` and `AlmanacRegistration` (`@derivedFrom(field: "record")`):

```
ledger-subquery-subquery-node-1  | 2022-11-21T14:06:05.297Z <store> ERROR Having a probl
em when syncing schema SequelizeDatabaseError: syntax error at or near "'almanac_registr
ations_record_id_uindex'"   
```

**Comment by bryanchriswhite** [comment]
Waiting for the v0.2 changes.

**Comment by bryanchriswhite** [comment]
Sprint 12 -> 13 update:

- [ ] Dependency permission issues
  - [x] submodule access
  - [ ] private repo release asset
- [x] Migration debugging
- [x] Logging woes (see #223)
